eng/common/scripts/SemVer.ps1

```diff
@@ -31,9 +52,18 @@ class SemVer {
                 $prelabel = $matches["prelabel"]
                 $prenumber = [int]$matches["prenumber"]
                 $isPre = $true;
+            
+                if ($enforcePrereleaseLabelSemantics -eq $true){
+                    if ($prelabel -ne 'preview') {
```

Comment:
I suggest we put this in a separate method. That folks can call if they wish to verify the and they are willing to handle the potential exceptions.

Reply:
Pulled out